app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -107,11 +110,33 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
     fun releasePlayer() {
         mediaSession.isActive = false
         mediaSession.release()
-        playerOrNull?.let { player ->
+        val playerState = playerOrNull?.let { player ->
+            val state = player.playbackState to player.currentPosition
             player.removeListener(this)
             player.release()
+            state
         }
         _player.value = null
+
+        val mediaSource = mediaSourceManager.jellyfinMediaSource.value
+        if (playerState != null && mediaSource != null) {
+            // viewModelScope is already cancelled at this point, so we need a fallback
```

Comment:
I think you can fix this by calling `releasePlayer` in the `onDestroy` function instead of `onCleared`.

![image](https://user-images.githubusercontent.com/2305178/92611738-64121680-f2b9-11ea-8964-1e59ef9882f4.png)

Reply:
And what about the lifecycle observer (PlayerLifecycleObserver)? Won't that one trigger before the onCleared?

Reply:
It'd also trigger on every `onDestroy` from the activity afaik.

Reply:
Interestingly, no. The (Player)LifecycleObserver correctly handles rotation events and e.g. only dispatches `onCreate` and `onStop` once, but onDestroy interestingly never fires. I assume that the ViewModel gets destroyed before the activity.